packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -484,8 +484,10 @@ class ScaffoldState extends State<Scaffold> {
       List<Widget> bottomSheets = <Widget>[];
       if (_dismissedBottomSheets != null && _dismissedBottomSheets.isNotEmpty)
         bottomSheets.addAll(_dismissedBottomSheets);
-      if (_currentBottomSheet != null)
+      if (_currentBottomSheet != null) {
+        _currentBottomSheet.setState(() { });
```

Comment:
but maybe @abarth has an idea. The problem here is that the scaffold wants to make sure that its bottom sheet's builder is invoked whenever it is itself invoked, but in between the scaffold and the builder there's another widget whose description we've pre-built and keep inserting (so it terminates the build walk and the builder never gets called).